plugins/fileservices/fileservices.cpp

```diff
@@ -3008,3 +3011,46 @@ FILESERVICES_API void FILESERVICES_CALL fsGetDropZones(ICodeContext *ctx, size32
     __lenResult = mb.length();
     __result = mb.detach();
 }
+
+
+
+FILESERVICES_API int FILESERVICES_CALL fsGetExpireDays(ICodeContext * ctx, const char *_lfn)
+{
+    StringBuffer lfn;
+    constructLogicalName(ctx, _lfn, lfn);
+    Linked<IUserDescriptor> udesc = ctx->queryUserDescriptor();
+    Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(lfn.str(),udesc, false,false, false, nullptr, defaultPrivilegedUser);
+    int expireDays = -1;
```

Comment:
trivial: but now you're not filling a reference, no point in this line, may as well just return, i.e.:
```
   if (df)
       return df->getExpire();
   else
        throw makeStringExceptionV(0, "GetExpireDays: Could not find logical file %s", lfn.str());
}
```

Reply:
Changed